app/forms/claim_form.rb

```diff
@@ -0,0 +1,45 @@
+class ClaimForm
+  include ActiveModel::Model
+  STEPS = %w[providers mentors].freeze
+
+  attr_accessor :claim, :step, :claim_params
+
+  def save
+    claim.assign_attributes(claim_params)
+    return false unless valid?
+
+    claim.save!
+  end
+
+  def valid?
+    claim.valid? && valid_providers? && valid_mentors?
+  end
+
+  private
+
+  def valid_providers?
+    return true unless step == "providers"
+
+    claim.mentor_trainings.each do |mentor_training|
+      if mentor_training.provider_id.nil?
+        mentor_training.errors.add(:provider_id, :blank)
+        claim.errors.add(:base, :enter_provider_name)
+      end
+    end
+
+    claim.errors.blank?
+  end
+
+  def valid_mentors?
+    return true unless step == "mentors"
+
+    claim.mentor_trainings.each do |mentor_training|
+      if mentor_training.teacher_id.nil?
+        mentor_training.errors.add(:teacher_id, :blank)
+        claim.errors.add(:base, :enter_teacher_name)
```

Comment:
This might spit out too many errors if we allow the user to add multiple mentors at the same time. At this point though, we are not sure if the user can add multiple mentors or just one, or the process might be different. So the idea is to leave it as it is and come back when we have more details.